src/harness/testcases/WINNF_FT_S_PPR_testcase.py

```diff
@@ -401,3 +409,290 @@ def test_WINNF_FT_S_PPR_2(self, config_filename):
+    device_d['installationParam']['latitude'] = 38.846125
+    device_d['installationParam']['longitude'] = -97.156184
+    device_d['installationParam']['antennaBeamwidth'] = 0
+    device_d['installationParam']['antennaDowntilt'] = 0
+    device_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_e.json')))
+    device_e['installationParam']['latitude'] = 38.761748
+    device_e['installationParam']['longitude'] = -97.118459
+
+    # Pre-load conditionals and remove REG-conditional fields from registration
+    # requests.
+    conditional_keys = [
+        'cbsdCategory', 'fccId', 'cbsdSerialNumber', 'airInterface',
+        'installationParam', 'measCapability'
+    ]
+    reg_conditional_keys = [
+        'cbsdCategory', 'airInterface', 'installationParam', 'measCapability'
+    ]
+    conditionals_b = {key: device_b[key] for key in conditional_keys}
+    device_b = {
+        key: device_b[key]
+        for key in device_b
+        if key not in reg_conditional_keys
+    }
+    conditionals_d = {key: device_d[key] for key in conditional_keys}
+    device_d = {
+        key: device_d[key]
+        for key in device_d
+        if key not in reg_conditional_keys
+    }
+
+    # Load grant requests (default is 3620-3630).
+    grant_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_b['operationParam']['maxEirp'] = 30
+    grant_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_d = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_d['operationParam']['maxEirp'] = 30
+    grant_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+
+    # CBSDs in SAS UUT.
+    domain_proxy = {
+        'registrationRequests': [device_b, device_d],
+        'grantRequests': [grant_b, grant_d],
+        'conditionalRegistrationData': [conditionals_b, conditionals_d],
+        'cert': getCertFilename('domain_proxy.cert'),
+        'key': getCertFilename('domain_proxy.key')
+    }
+
+    # One PPA in SAS UUT.
+    pal_low_frequency = 3620000000
+    pal_high_frequency = 3630000000
+    pal_record_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+    ppa_record_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+    ppa_record_0['zone']['features'][0]['geometry']['coordinates'] = [[[
+        -97.155, 38.75
+    ], [-97.155, 38.85], [-97.165, 38.85], [-97.165, 38.75], [-97.155, 38.75]]]
+    ppa_record_0, pal_records_0 = makePpaAndPalRecordsConsistent(
+        ppa_record_0, [pal_record_0], pal_low_frequency, pal_high_frequency,
+        'test_user_1')
+    device_b['userId'] = 'test_user_1'
+    ppa_cluster_list = [0]  # The first device is device B.
```

Comment:
Assuming both deviceB and deviceD belong to the SUUT PPA and will get PAL grants we would need to add deviceD to the clusterlist, right? or are we trying to have deviceD fail to get a grant?

Reply:
Javi,
as I read in the test comment (line 418):
SAS UUT has one PPA, with device B as the only CBSD.

so I think that  was done on purpose.

Reply:
It was originally done on purpose, but now it's wrong. Thank you and fixed!